utkdigitalinitiatives/utk_lib_digital_front
# Add Cache busting to digital front React app

**What does this Pull Request do?**

Integrates basic cache busting for app generated JS and CSS by adding a unique hashed pattern for each build. Outputs hashed file assets and corresponding HTML structure for index.html on new build. Webpack is used for generation of the production and test files and thus new plugins were added and/or updated to facilitate the build process.

**How should this be tested?**

1. `git clone <repo>`
2. `cd <repo>`
3. `yarn`
4. `git pull`
5. `git checkout <branch>`
6. `yarn build`
7. View source of `./index.html`, networked assets for digital specific compiled JavaScript and CSS will be hashed uniquely. Check `./dist/digital.*` mappings and make sure that the file references follow correctly.
9. Open the generated root `./index.html` in your browser. App should render as expected.

**Additional Notes:**
Steps 1-3 are only required if this is the first time running the app locally. The root `./index.html` has now been ignore from the repo as this file is generated on build from `./src/index.html`

**Interested parties**
@DonRichards
